ObjectiveCloudantTests/QueryViewTests.m

```diff
@@ -0,0 +1,966 @@
+//  and limitations under the License.
+//
+
+#import <XCTest/XCTest.h>
+#import <ObjectiveCloudant/ObjectiveCloudant.h>
+#import "TestHelpers.h"
+
+@interface QueryViewTests : XCTestCase
+@property (nonatomic, strong) NSString *url;
+@property (nonatomic, strong) NSString *dbName;
+@property (nonatomic, strong) NSString *username;
+@property (nonatomic, strong) NSString *password;
+@property (nonatomic, strong) CDTDatabase *database;
+@end
+
+@implementation QueryViewTests
+
+- (void)setUp
+{
+    [super setUp];
+
+    self.url = @"http://localhost:5984";
+    self.username = nil;
+    self.password = nil;
+
+    // These tests require their own database as they modify content; create one
+
+    self.dbName = @"objectivecouch-test";
+
+    CDTCouchDBClient *client = [CDTCouchDBClient clientForURL:[NSURL URLWithString:self.url]
+                                                     username:self.username
+                                                     password:self.password];
+    self.database = client[self.dbName];
+}
+
+- (void)tearDown
+{
+    // Put teardown code here. This method is called after the invocation of each test method in the
+    // class.
+    [super tearDown];
+}
+
+- (void)testfailsWithKeyandKeysSet
+{
+    XCTestExpectation *completionFailed =
+        [self expectationWithDescription:@"Keys and Key property set"];
+
+    CDTQueryViewOperation *op = [[CDTQueryViewOperation alloc] init];
+    op.ddoc = @"views101";
+    op.viewName = @"diet";
+    op.key = @"key";
+    op.keys = @[ @"keys" ];
+
+    op.queryViewCompletionBlock = ^(NSError *error) {
+      [completionFailed fulfill];
+      XCTAssertNotNil(error);
+    };
+
+    [self.database addOperation:op];
+
+    [self waitForExpectationsWithTimeout:10.0f
+                                 handler:^(NSError *_Nullable error) {
+                                   NSLog(@"Operation completion handler not called");
+                                 }];
+}
+
+- (void)testfailsMissingDdoc
+{
+    XCTestExpectation *completionFailed = [self expectationWithDescription:@"missing ddoc"];
+
+    CDTQueryViewOperation *op = [[CDTQueryViewOperation alloc] init];
+    op.viewName = @"diet";
+    op.key = @"key";
+    op.keys = @[ @"keys" ];
+
+    op.queryViewCompletionBlock = ^(NSError *error) {
+      [completionFailed fulfill];
+      XCTAssertNotNil(error);
+    };
+
+    [self.database addOperation:op];
+
+    [self waitForExpectationsWithTimeout:10.0f
+                                 handler:^(NSError *_Nullable error) {
+                                   NSLog(@"Operation completion handler not called");
+                                 }];
+}
+
+- (void)testfailsMissingViewNAME
+{
+    XCTestExpectation *completionFailed = [self expectationWithDescription:@"Missing view name"];
+
+    CDTQueryViewOperation *op = [[CDTQueryViewOperation alloc] init];
+    op.ddoc = @"views101";
+    op.viewName = @"diet";
+    op.key = @"key";
+    op.keys = @[ @"keys" ];
```

Comment:
We shouldn't set `op.key` and `op.keys` or the test might pass for the wrong reason.